### PR TITLE
feat: query tab for insights

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

see https://github.com/PostHog/posthog/issues/13927

## Changes

* Add a "query" tab to the insight page if data exploration is on.
* Allow saving and updating insights

NB requires two flags active
1. DATA_EXPLORATION_QUERIES_ON_DASHBOARDS
2. DATA_EXPLORATION_INSIGHTS

## on the saved insights page

savedinsightslist

## on the insights page

For follow-up work

* add a query editor directly in the insight edit view
* proper summary of query insights in titles and show details
* filtering of query insights in the saved insight list
* draw the rest of the owl as listed in https://github.com/PostHog/posthog/issues/13927

## How did you test this code?

a few developer tests and 👀

### CODE REVIEW

File: frontend/src/scenes/insights/insightLogic.ts [hunk 1]
```diff
@@ -209,13 +209,14 @@ export const insightLogic = kea<insightLogicType>([
...
 
-                    if ('filters' in insight && emptyFilters(insight.filters)) {
+                    if ('filters' in insight && !insight.query && emptyFilters(insight.filters)) {
                         const error = new Error('Will not override empty filters in updateInsight.')
                         Sentry.captureException(error, {
                             extra: {
                                 filters: JSON.stringify(insight.filters),
                                 insight: JSON.stringify(insight),
                                 valuesInsight: JSON.stringify(values.insight),
+                                query: JSON.stringify(insight.query),
```
Comment (thmsobrmlr) [inline frontend/src/scenes/insights/insightLogic.ts:219]: This doesn't do anything with the `!insight.query` clause in the if-statement before.

File: frontend/src/scenes/insights/insightLogic.test.ts [hunk 2]
```diff
@@ -1276,4 +1300,43 @@ describe('insightLogic', () => {
...
+                    { filters: {}, query: { kind: NodeKind.DataTableNode } as DataTableNode },
+                    { overrideFilter: true }
+                )
+                logic.actions.saveInsight()
+            })
+
+            const mockCreateCalls = (api.create as jest.Mock).mock.calls
+            // there will be at least two used client query ids
+            // the most recent has not been cancelled
+            // the one before that has been
```
Comment (thmsobrmlr) [inline frontend/src/scenes/insights/insightLogic.test.ts:1326]: I think this comment is copypasta from somewhere else? Or else I don't understand it :D

**Review by thmsobrmlr — APPROVED** [review]
Looks good! Left some small observations inline.

File: posthog/api/insight.py [hunk 3]
```diff
@@ -564,9 +564,9 @@ def get_queryset(self) -> QuerySet:
         if self.action == "list":
             queryset = queryset.prefetch_related("tagged_items__tag")
             queryset = self._filter_request(self.request, queryset)
-            # exclude insights that have a query but no filters (for now)
-            if not self.request.query_params.get("include_query_insights", False):
-                queryset = queryset.exclude(filters={})
+
+            if self.request.query_params.get("include_query_insights", "false").lower() != "true":
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply posthog/api/insight.py:568]: I think we're running before serializers etc kick in here so I don't think there's a better way (although I'm learning Django/DRF as I go so could be wrong)

But we'll delete this when the query switcheroo is complete so...

**Pull request merged**